baselines/niid_bench/README.md

```diff
@@ -0,0 +1,108 @@
+
+| Datasets | #classes | #partitions | partitioning method | partition settings |
+| :------ | :---: | :---: | :---: | :---: |
+| CIFAR10, MNIST, Fashion-mnist | 10 | 10 | IID<br>dirichlet<br>sort and partition<br>sort and partition<br>sort and partition | NA<br>distribution parameter 0.5<br>1 label per client<br>2 labels per client<br>3 labels per client |
+
+
+****Training Hyperparameters:**** There are four FL algorithms and they have many common hyperparameters and few different ones. The following table shows the common hyperparameters and their default values.
+
+| Description | Default Value |
+| ----------- | ----- |
+| total clients | 10 |
+| clients per round | 10 |
+| number of rounds | 50 |
+| number of local epochs | 10 |
+| client resources | {'num_cpus': 4.0, 'num_gpus': 0.0 }|
+| dataset name | cifar10 
+| data partition | Dirichlet (0.5) |
+| batch size | 64 |
+| momentum for SGD | 0.9 |
+
+For FedProx algorithm the proximal parameter is tuned from values {0.001, 0.01, 0.1, 1.0} in the experiments. The default value is 0.01. 
+
+
+## Environment Setup
+
+```bash
+# Setup the base poetry enviroment from the niid_bench directory
+# Set python version
+pyenv local 3.10.6
+# Tell poetry to use python 3.10
+poetry env use 3.10.6
+# Now install the environment
+poetry install
+# Start the shell
+poetry shell
+```
+
+
+## Running the Experiments
+You can run four algorithms fedavg, scaffold, fedprox, and fednova. To run any of them, use any of the corresponding config files. For instance, the following command will run with the default config provided in the corresponding configuration files.
+
+```bash
+# Run with default config, it will run FedAvg on cpu-only mode
+python -m niid_bench.main
+# Below to enable GPU utilization by the server and the clients.
+python -m niid_bench.main server_device=cuda client_resources.num_gpus=0.2
+```
+
+To change the configuration such as dataset or hyperparameters, specify them as part of the command line arguments.
+
+```bash
+python -m niid_bench.main --config-name scaffold_base dataset_name=mnist partitioning=iid # iid
+python -m niid_bench.main --config-name fedprox_base dataset_name=mnist partitioning=dirichlet # dirichlet
+python -m niid_bench.main --config-name fednova_base dataset_name=mnist partitioning=label_quantity labels_per_client=3 # sort and partition
+```
+
+
+## Expected Results
+
+We provide the bash script run_exp.py that can be used to run all configurations. For instance, the following command runs all of them with 7 configurations running at the same time.
+
+```bash
+python run_exp.py --seed 42 --num-processes 7
+```
+
+The above command generates results that can be parsed to get the accuracies for each configuration which can be presented in a table (similar to Table 3 in the paper).
```

Comment:
After running your experiments I noticed in several cases (which is expected I guess) not the best performance was achieved in the last round. I assume you reported the best observed result (then averaged those over the several seeds you used). I think it's worth indicating this (in case people only look into the last round acc)